pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1914,4 +1914,13 @@ public void shutdownNow() {
     protected BrokerService newBrokerService(PulsarService pulsar) throws Exception {
         return new BrokerService(pulsar, ioEventLoopGroup);
     }
+
+    private CompactionServiceFactory loadCompactionServiceFactory() {
+        String compactionServiceFactoryClassName = config.getCompactionServiceFactoryClassName();
+        var compactionServiceFactory =
+                Reflections.createInstance(compactionServiceFactoryClassName, CompactionServiceFactory.class,
+                        Thread.currentThread().getContextClassLoader());
+        compactionServiceFactory.initialize(this).join();
```

Comment:
Well, I am wondering if it's valuable to use get with a timeout to avoid the broker hanging forever in some thread-blocking cases. :)

Reply:
+1, Why do we let the initialize method return `CompletableFuture`? Do you think we can make it sync?

Reply:
@coderzc Maybe you missed this one

Reply:
I think the asynchronous interface is more extensible, and if we support asynchronous initialize in the PulsarService in the future, we don't need to change the interface